console-subscriber/src/builder.rs

```diff
@@ -376,17 +424,19 @@ impl Builder {
         thread::Builder::new()
             .name("console_subscriber".into())
             .spawn(move || {
-                let runtime = runtime::Builder::new_current_thread()
-                    .enable_io()
-                    .enable_time()
-                    .build()
-                    .expect("console subscriber runtime initialization failed");
-
-                runtime.block_on(async move {
-                    server
-                        .serve()
-                        .await
-                        .expect("console subscriber server failed")
+                tracing::subscriber::with_default(NoNoSubscriber::default(), || {
+                    let runtime = runtime::Builder::new_current_thread()
+                        .enable_io()
+                        .enable_time()
+                        .build()
+                        .expect("console subscriber runtime initialization failed");
+
+                    runtime.block_on(async move {
+                        server
+                            .serve()
+                            .await
+                            .expect("console subscriber server failed")
+                    });
```

Comment:
nit, take it or leave it: we could avoid adding an additional level of indentation if we used [`tracing::subscriber::set_default`](https://docs.rs/tracing/latest/tracing/subscriber/fn.set_default.html), which returns a `Drop` guard, rather than `with_default` here:
```suggestion
                let _subscriber = tracing::subscriber::set_default(NoNoSubscriber::default());
                let runtime = runtime::Builder::new_current_thread()
                    .enable_io()
                    .enable_time()
                    .build()
                    .expect("console subscriber runtime initialization failed");

                runtime.block_on(async move {
                    server
                        .serve()
                        .await
                        .expect("console subscriber server failed")
```

I have a slight preference for doing it this way, because it makes the Git diff for this change more focused (currently, the indentation change makes all the code inside the `with_default` closure appear to be "added" in this PR). But, it's not a big deal, and I won't block merging on that.

Reply:
Git has a flag for that, GitHub doesn't.
Changed, the guard works better when setting the subscriber is conditional.